src/Features/Core/Portable/CodeRefactorings/SyncNamespace/IChangeNamespaceService.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.ChangeNamespace
+{
+    internal interface IChangeNamespaceService : ILanguageService
+    {
+        Task<Solution> ChangeNamespaceAsync(Solution solution, ImmutableArray<DocumentId> documentIds, string declaredNamespace, string targetNamespace, CancellationToken cancellationToken);
+
+        bool TryGetReplacementReferenceSyntax(SyntaxNode reference, ImmutableArray<string> newNamespaceParts, ISyntaxFactsService syntaxFacts, out SyntaxNode old, out SyntaxNode @new);
```

Comment:
this feels lke this should not be in the interface.  It's only needed by the service internally right? So why not just have it be in the abstract class?

Reply:
effectively, i think this would be split out separately.  perhaps another internal interface.  The main interface which we might expect anyone to consume would just have the high level operation they call on the Solution.  That impl would then call into internal interfaces which could do the work.

Reply:
Completely agree, this should be hidden from the service interface.